pydantic/env_settings.py

```diff
@@ -217,7 +217,19 @@ def field_is_complex(self, field: ModelField) -> Tuple[bool, bool]:
 
         return True, allow_json_failure
 
-    def explode_env_vars(self, field: ModelField, env_vars: Mapping[str, Optional[str]]) -> Dict[str, Any]:
+    @staticmethod
+    def next_field(field: Optional[ModelField], key: str) -> Optional[ModelField]:
+        if not field or field.sub_fields:
```

Comment:
This method is only called in one place, are you sure it needs to be a separate function?

If so, please add a docstring explaining what we're doing here.

Reply:
This method gets called twice, that's why I moved it to a method:

https://github.com/pydantic/pydantic/pull/4216/files/82f5b971d3a201fea1402eb5ae8b210a03ad5e2a#diff-67b5bf5da243ce3b23618214851e35a3f0951de739f3747800daf6d1273e4962R248

https://github.com/pydantic/pydantic/pull/4216/files/82f5b971d3a201fea1402eb5ae8b210a03ad5e2a#diff-67b5bf5da243ce3b23618214851e35a3f0951de739f3747800daf6d1273e4962R252

Reply:
Sorry, you're right, my mistake.

Reply:
I don't think there are collisions with existing setting schemas due to the fact that setting complex values from an environment variable never really worked.

Nevertheless, it's up to you - for now I'm using a monkeypatch that does the job pretty well for me.

Reply:
I understand where you're coming from, I'm just a bit paranoid at the moment about "breaking" changes, hence deferring this.